### DISCUSSION TITLE
Dynamic hosts

### DISCUSSION THREAD

**Discussion by konradmichalik** [deployphp/deployer#3495] [General]
Hej there,

i'm working on a workflow to implement some kind of a feature branch deployment. Therefore I want to create a dynamic host based on a existing "meta" host on runtime regarding CLI arguments, so running for example the following command:

```bash
vendor/bin/dep deploy stage --feature=custom-feature-branch-name
```

With this, for example the deploy path has to be changed to `get('deploy_path') . input()->getOption('feature')`.

I know, I can handle this by my own using a init task for extending the deployment process:

```php
before('deploy', 'deploy:init');
```

But this seems to become tricky, when running single commands, e.g. `vendor/bin/dep deploy:unlock stage --feature=custom-feature-branch-name`.

Is there a possibility to hook in the host definition to handle them globally? 

Thanks so far, keep up the good work.

**Comment by antonmedv** [comment]
As recipes are just simple PHP code you can do anything.

```php
if (getenv('FEATURE_HOST')) {
    host(getenv('FEATURE_HOST'))
        ->setDeployPath('~/path/' . getenv('FEATURE_HOST'))
...
}
```

**Reply by konradmichalik** [reply]
Interesting, so what would be the best way to copy the configuration from an existing host?

```php
$baseHost = Deployer::get()->hosts->get('stage');
$baseHost->setDeployPath($baseHost->getDeployPath() . getenv('FEATURE_HOST'));

Deployer::get()->hosts->set(getenv('FEATURE_HOST'), $baseHost);
```

And calling them like this:
```bash
FEATURE_HOST=custom-feature-branch-name vendor/bin/dep deploy custom-feature-branch-name
```

**Reply by antonmedv** [reply]
Just put configuration not in the host but at top level of the recipe. It will be inherited. https://deployer.org/docs/7.x/basics
